### PR TITLE
Tickets/dm38531: Better script timeout handling.

### PR DESCRIPTION
**Pull request by rbovill**

### CODE REVIEW

File: python/lsst/ts/IntegrationTests/base_script.py [hunk 1]
```diff
@@ -117,35 +131,26 @@ def add_arguments(cls, **kwargs: str) -> None:
...
         """
-        while True:
-            data = await scriptqueue_remote.evt_script.next(
-                flush=False, timeout=timeout
-            )
-            if data.scriptSalIndex != sal_index:
-                continue
-            if data.scriptState in self.terminal_states:
-                return data.scriptState
+        print(type(data))
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/base_script.py:142]: Probably can get rid of this now.

File: python/lsst/ts/IntegrationTests/base_script.py [hunk 2]
```diff
@@ -117,35 +131,26 @@ def add_arguments(cls, **kwargs: str) -> None:
...
+        if data.scriptState in self.terminal_states:
+            print("Script done.")
+            # Store the final script state in the script_states list.
+            self.script_states.append(int(data.scriptState))
+            # Scripts run sequentially and FIFO.
+            # When done, remove the leading script.
+            self.temp_script_indexes.pop(0)
+            # Set the all_scripts_done flag to True when all the
+            # scripts are complete.
+            self.all_scripts_done = bool(len(self.temp_script_indexes) == 0)
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/base_script.py:153]: `bool` cast isn't necessary. `==` returns a `bool`

File: python/lsst/ts/IntegrationTests/base_script.py [hunk 3]
```diff
@@ -25,6 +25,7 @@
 from lsst.ts.idl.enums.Script import ScriptState
 
 from datetime import date
+import copy
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/base_script.py:28]: We should have `isort` setup for this repo.

**Review by mareuter — APPROVED** [review]
A few things to fix up, but overall this looks good.

File: python/lsst/ts/IntegrationTests/base_script.py (hunk 3 above)
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/base_script.py:28]: I have a task for next sprint to implement pre_commit and stuff on this package.

**Pull request merged**